client/js/main.ts

```diff
@@ -403,48 +509,106 @@ document.getElementById("add-new-tag")!.addEventListener("click", e => {
 	button.disabled = true;
 
 	let tagInput = <HTMLInputElement> document.getElementById("new-tag-name");
-	let currentTagSelect = <HTMLSelectElement> document.getElementById("current-tag");
-
-	let currentTag: string = currentTagSelect.options[currentTagSelect.selectedIndex].value;
-	if (!currentTag) {
-		alert("Please select a valid tag");
-		button.disabled = false;
-		return; 
-	}
 	let tag: string = tagInput.value.trim().toLowerCase();
 	if (!tag) {
 		alert("Please enter a tag name");
 		button.disabled = false;
 		return;
 	}
-	qwest.put(`/api/data/addTag/${tag}`, {
-		currentTag: currentTag
-	}).then((xhr, response) => {
-		let tags: string[] = Array.prototype.slice.call(document.querySelectorAll("#tag-choose > option")).map((el: HTMLOptionElement) => el.textContent );
-		// Add to tag selectors
-		if (tags.indexOf(tag) === -1) {
-			const tagsList = document.querySelectorAll("select.tags");
-			Array.prototype.slice.call(tagsList).forEach((el: HTMLSelectElement) => {
-				let option = document.createElement("option");
-				option.textContent = tag;
-				el.appendChild(option);
-			});
-		}
-
-		// Clear the form
-		tagInput.value = tagInput.defaultValue;
-		currentTagSelect.options[0].selected = true;
 
+	qwest.post(`/graphql`, JSON.stringify({
+		query: `mutation {
+			add_tag(tag: "${tag}") {
+				name
+			}
+		}`
+	}), graphqlOptions).then((xhr, response) => {
+		// Add to tag selectors
+		updateTagSelectors([tag]);
+		
+		// Clear form
+		tagInput.value = "";
 		document.querySelector(`label[for="new-tag-name"]`)!.classList.remove("mdc-textfield__label--float-above");
-
+		
 		alert("Successfully added tag to attendee(s)!");
 	}).catch((e, xhr, response) => {
-		alert(response.error);
+		console.log(response);
+		alert("An error occurred while adding the tag");
 	}).complete(() => {
 		button.disabled = false;
 	});	
 });
 
+// Populate checkboxes for question names
+qwest.post("/graphql", JSON.stringify({
+	query: `{ question_names }`
+}), graphqlOptions).then((xhr, response) => {
+	let checkboxTemplate = <HTMLTemplateElement> document.getElementById("checkbox-item")!;
+	let checkboxContainer = document.getElementById("question-options")!;
+	let button = document.getElementById("button-row")!;
+	if (!response.data || !response.data.question_names) {
+		return;
+	}
+
+	let question_names: string[] = response.data.question_names.sort((a: string, b: string) => {
+		return a.localeCompare(b);
+	});
+
+	for (let curr of question_names) {
+		let node = document.importNode(checkboxTemplate.content, true) as DocumentFragment;
+		node.querySelector("input")!.id = curr;
```

Comment:
Cache this in a separate variable so that you don't have to keep running `.querySelector()`.